src/python/ensembl/utils/argparse.py

```diff
@@ -0,0 +1,142 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Provide an extended version of :class:`argparse.ArgumentParser` with additional functionality."""
+
+__all__ = ["ArgumentParser"]
+
+import argparse
+import os
+from pathlib import Path
+
+import sqlalchemy
+
+
+def _validate_src_path(parser: argparse.ArgumentParser, src_path: os.PathLike) -> Path:
+    """Returns the path if exists and it is readable, raises an error through the argument parser otherwise.
+    
+    Args:
+        parser: Argument parser.
+        src_path: File or directory path to check.
+
+    """
+    src_path = Path(src_path)
+    if not src_path.exists():
+        parser.error(f"'{src_path}' not found")
+    elif not os.access(src_path, os.R_OK):
+        parser.error(f"'{src_path}' not readable")
+    return src_path
+
+
+def _validate_dst_path(parser: argparse.ArgumentParser, dst_path: os.PathLike) -> Path:
+    """Returns the path if it is writable, raises an error through the argument parser otherwise.
+    
+    Args:
+        parser: Argument parser.
+        dst_path: File or directory path to check.
+
+    """
+    dst_path = Path(dst_path)
+    if dst_path.exists() and os.access(dst_path, os.W_OK):
+        return dst_path
+    for parent_path in dst_path.parents:
+        if parent_path.exists() and not os.access(parent_path, os.W_OK):
+            parser.error(f"'{dst_path}' is not writable")
+    return dst_path
```

Comment:
Do we want to add an option to fail if the dst file already exists? Or is that too much to do at this stage